### ISSUE TITLE
[4.x]: Debug bar errors

### ISSUE THREAD

**Issue by hassanmah** [craftcms/cms#12010]
### Description
On some pages, I get errors when using the debug bar on all tabs

On Events and DB

```
yii\base\ErrorException: Trying to access array offset on value of type null in /deploy/vendor/yiisoft/yii2-debug/src/views/default/panels/request/summary.php:8
Stack trace:
#0 /deploy/vendor/craftcms/cms/src/web/ErrorHandler.php(79): yii\base\ErrorHandler->handleError(2, 'Trying to acces...', '/Users/hassanma...', 8)
#1 /deploy/vendor/yiisoft/yii2-debug/src/views/default/panels/request/summary.php(8): craft\web\ErrorHandler->handleError(2, 'Trying to acces...', '/Users/hassanma...', 8)
#2 /deploy/vendor/yiisoft/yii2/base/View.php(347): require('/Users/hassanma...')
#3 /deploy/vendor/yiisoft/yii2/base/View.php(257): yii\base\View->renderPhpFile('/Users/hassanma...', Array)
#4 /deploy/vendor/yiisoft/yii2/base/View.php(156): yii\base\View->renderFile('/Users/hassanma...', Array, NULL)
#5 /deploy/vendor/yiisoft/yii2-debug/src/panels/RequestPanel.php(43): yii\base\View->render('panels/request/...', Array)
#6 /deploy/vendor/yiisoft/yii2-debug/src/views/default/view.php(26): yii\debug\panels\RequestPanel->getSummary()
#7 /deploy/vendor/yiisoft/yii2/base/View.php(347): require('/Users/hassanma...')
#8 /deploy/vendor/yiisoft/yii2/base/View.php(257): yii\base\View->renderPhpFile('/Users/hassanma...', Array)
#9 /deploy/vendor/yiisoft/yii2/base/View.php(156): yii\base\View->renderFile('/Users/hassanma...', Array, Object(yii\debug\controllers\DefaultController))
#10 /deploy/vendor/yiisoft/yii2/base/Controller.php(407): yii\base\View->render('view', Array, Object(yii\debug\controllers\DefaultController))
#11 /deploy/vendor/yiisoft/yii2-debug/src/controllers/DefaultController.php(126): yii\base\Controller->render('view', Array)
#12 [internal function]: yii\debug\controllers\DefaultController->actionView('63318b520d4ca', 'event')
#13 /deploy/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#14 /deploy/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#15 /deploy/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('view', Array)
#16 /deploy/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('debug/default/v...', Array)
#17 /deploy/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('debug/default/v...', Array)
#18 /deploy/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#19 /deploy/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#20 /deploy/public/index.php(12): yii\base\Application->run()
#21 /Users/.composer/vendor/laravel/valet/server.php(234): require('/Users/hassanma...')
#22 {main}
```

On Request:

```

TypeError: Cannot assign yii\debug\FlattenException to property craft\events\ExceptionEvent::$exception of type Throwable in /deploy/vendor/yiisoft/yii2/BaseYii.php:558
Stack trace:
#0 /deploy/vendor/yiisoft/yii2/base/BaseObject.php(107): yii\BaseYii::configure(Object(craft\events\ExceptionEvent), Array)
#1 /deploy/vendor/craftcms/cms/src/web/ErrorHandler.php(49): yii\base\BaseObject->__construct(Array)
#2 /deploy/vendor/yiisoft/yii2-debug/src/controllers/DefaultController.php(118): craft\web\ErrorHandler->handleException(Object(yii\debug\FlattenException))
#3 [internal function]: yii\debug\controllers\DefaultController->actionView('63318b520d4ca', 'request')
#4 /deploy/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#5 /deploy/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#6 /deploy/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('view', Array)
#7 /deploy/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('debug/default/v...', Array)
#8 /deploy/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('debug/default/v...', Array)
#9 /deploy/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#10 /deploy/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#11 /deploy/public/index.php(12): yii\base\Application->run()
#12 /Users/.composer/vendor/laravel/valet/server.php(234): require('/Users/hassanma...')
#13 {main}
```

### Steps to reproduce

1. Open debug bar

### Craft CMS version

4.2.5.1

### Installed plugins and versions

-

**Comment by angrybrad** [comment]
Probably a Yii2 issue? https://github.com/yiisoft/yii2-debug/issues/491

**Comment by hassanmah** [comment]
@brianjhanson just a Neo field

**Comment by 730projects** [comment]
I kind of figured it out for me. I'm using the [Typed Link Field plugin](https://plugins.craftcms.com/typedlinkfield). If I remove all of the instances of that field from my Matrix Field the error disappears. After re-adding the Typed Link fields one by one the error didn't return. Bit of a weird bug, but I think I had some misconfiguration in one of the Typed Link fields...

**Comment by brianjhanson** [comment]
I'm still having trouble replicating this consistently. If anyone has some consistent steps to reproduce that would be great. 

Based on the stack trace it looks like in some situations an instance of `ExceptionEvent` is getting created with an instance of `yii\debug\FlattenException` which isn't compatible with the `$exception` property there. I'm guessing this has something to do with a plugin, but I've had trouble replicating it consistently.

**Comment by 730projects** [comment]
The error was way easier for me than I initially thought ... I had a custom module that wasn't up to speed with Craft's coding standards. After fixing the module everything worked fine again 😅 Do you have any custom modules/plugins? @hassanmah

**Comment by brianjhanson** [comment]
Glad you got it sorted out! I'm going to close this for now, but @hassanmah feel free to reopen if you're still running into issues.

**Comment by hassanmah** [comment]
@daafbleumink sorry for the reply, I do have a custom module, which I deletes the folder for, but seems like it's still in composer.json

Do I need to run anything to remove it from composer.json?

Thanks

**Comment by brianjhanson** [comment]
@hassanmah If you're deleting the module, you'll want to manually remove the `autoload` config for your module from `composer.json` and then run `composer dump-autoload -a`.

**Comment by daafbleumink** [comment]
@hassanmah I think manually removing the entry from your composer.json, then running `composer install` + `composer dump-autoload` should work 🤔

**Comment by hassanmah** [comment]
@daafbleumink that's not fixed it, but looking at the comments, I do have [Typed Link Field plugin](https://plugins.craftcms.com/typedlinkfield) installed too.

**Comment by aaronbushnell** [comment]
FWIW I had the same issue and disabling/re-enabling the Typed Link Field plugin corrected this on my site!

**Comment by davidwebca** [comment]
I had this issue lately and I couldn't figure it out. I had used Typed Link Field before but since had migrated to Hyper from verbb and removed it. I also have a module, but it was existing prior to the issue. 

The thing that fixed it is restoring an old (but unused since 2 weeks ago) route in config/route.php. It was pointing to a controller in the module, but the route and controller were removed since a while ago. Restoring the route, refreshing the page, it worked... then I could remove the route again and it continued to work. Why the hell? I have no idea. 😂 It's as if an old cache pointing the route and controller together were failing when I was trying to access routes from debugbar...
